pkg/controller/instance/instance_controller.go

```diff
@@ -65,21 +65,95 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// reconcile (Instances). Specifically this calls for a reconciliation of any owned objects.
+	ovMapFn := handler.ToRequestsFunc(
+		func(a handler.MapObject) []reconcile.Request {
+			requests := make([]reconcile.Request, 0)
+			// We want to query and queue up operators Instances
+			instances := &kudov1alpha1.InstanceList{}
+			err := mgr.GetClient().List(
+				context.TODO(),
+				instances,
+				client.MatchingLabels(map[string]string{"operator": a.Meta.GetName()}),
+			)
+
+			if err != nil {
+				log.Printf("InstanceController: Error fetching instances list for operator %v: %v", a.Meta.GetName(), err)
+				return nil
+			}
+
+			for _, instance := range instances.Items {
+				// Sanity check - lets make sure that this instance references the operatorVersion
+				if instance.Spec.OperatorVersion.Name == a.Meta.GetName() &&
+					instance.GetOperatorVersionNamespace() == a.Meta.GetNamespace() &&
+					instance.Status.ActivePlan.Name == "" {
+
+					log.Printf("InstanceController: Creating a deploy execution plan for the instance %v", instance.Name)
+					err = createPlan(mgr, "deploy", &instance)
+					if err != nil {
+						log.Printf("InstanceController: Error creating \"%v\" object for \"deploy\": %v", instance.Name, err)
+					}
+
+					log.Printf("InstanceController: Queing instance %v for reconciliation", instance.Name)
+					requests = append(requests, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Name:      instance.Name,
+							Namespace: instance.Namespace,
+						},
+					})
+				}
+			}
+			log.Printf("Found %v instances to reconcile", len(requests))
+			return requests
+		})
+
+	// This map function makes sure that we *ONLY* handle created operatorVersion
+	ovPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+	}
+
+	if err = c.Watch(&source.Kind{Type: &kudov1alpha1.OperatorVersion{}}, &handler.EnqueueRequestsFromMapFunc{ToRequests: ovMapFn}, ovPredicate); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func instanceEventPredicateFunc(mgr manager.Manager) predicate.Funcs {
+	return predicate.Funcs{
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			old := e.ObjectOld.(*kudov1alpha1.Instance)
 			new := e.ObjectNew.(*kudov1alpha1.Instance)
 
 			// Get the OperatorVersion that corresponds to the new instance.
 			ov := &kudov1alpha1.OperatorVersion{}
-			err = mgr.GetClient().Get(context.TODO(),
+			err := mgr.GetClient().Get(context.TODO(),
 				types.NamespacedName{
 					Name:      new.Spec.OperatorVersion.Name,
 					Namespace: new.GetOperatorVersionNamespace(),
 				},
 				ov)
 			if err != nil {
-				log.Printf("InstanceController: Error getting operatorversion \"%v\" for instance \"%v\": %v",
+				log.Printf("InstanceEventPredicate: Error getting operatorversion \"%v\" for instance \"%v\": %v",
```

Comment:
I would leave the logging messages as they are, since they're still part of the instance controller.

Reply:
yeah... I thought about that and changed them based on tarun's original change... I'll change back based on your comment.